File: src/main/java/com/lsd/domain/scenario/Scenario.java
package com.lsd.domain.scenario;

import com.lsd.domain.scenario.diagram.SequenceDiagram;
import lombok.Builder;
import lombok.Value;

import java.util.Collection;

@Value
@Builder
public class Scenario {
    String title;
    String description;
    SequenceDiagram sequenceDiagram;
    Collection<Fact> facts;
}

File: src/main/kotlin/com/lsd/core/domain/Scenario.kt
package com.lsd.core.domain

data class Scenario
@JvmOverloads constructor(
    val title: String,
    val events: MutableList<SequenceEvent> = ArrayList(),
    val facts: MutableList<Fact> = ArrayList(),
    val description: String = "",
    val status: Status
)

enum class Status { ERROR, FAILURE, SUCCESS; }